mbuild/compound.py

```diff
@@ -1246,18 +1255,23 @@ def n_bonds(self):
             )
         return sum(1 for _ in self.bonds())
 
-    def add_bond(self, particle_pair):
+    def add_bond(self, particle_pair, bond_order=None):
         """Add a bond between two Particles.
 
         Parameters
         ----------
         particle_pair : indexable object, length=2, dtype=mb.Compound
             The pair of Particles to add a bond between
+        bond_order : float, optional, default=None
+            Bond order of the bond.
         """
         if self.root.bond_graph is None:
             self.root.bond_graph = BondGraph()
-
-        self.root.bond_graph.add_edge(particle_pair[0], particle_pair[1])
+        if bond_order is None:
+            bond_order = "default"
+        self.root.bond_graph.add_edge(
+            particle_pair[0], particle_pair[1], bo=bond_order
+        )
```

Comment:
Do you think we need a type check here to make sure the bond_order provided is of the right type (not bogus string)?

Reply:
Yeah, I think hard checking these would be useful. The options should be: 
"default", "single", "double", "triple", "aromatic", "unspecified". It seems the current options are:
"default", 1.0, 2.0, 3.0, 1.5, 0.0. 
I think being cleaner with using all strings would be nice, since we wouldn't have to deal with odd behavior for floats vs ints.

Also, these options should be all listed in the doc strings to_rdkit(), regardless of what they are.

Reply:
I added a check for `bond_order` using the types you suggested.